flight_backend/api/views/flight.py

```diff
@@ -0,0 +1,176 @@
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+from rest_framework.decorators import action
+from django.shortcuts import get_object_or_404
+
+from api.models import Aircraft, Location, Flight, Ticket
+from api.serializers import (
+    LocationSerializer,
+    AircraftSerializer,
+    FlightSerializer,
+    TicketSerializer,
+)
+from api.helpers import validate_flight_details
+from api.constants import STATUSES, AVAILABLE, BOOKED
+from api.tasks import flight_booking_notification, flight_reservation_notification
+
+
+class LocationViewSet(ModelViewSet):
+    serializer_class = LocationSerializer
+    queryset = Location.objects.all()
```

Comment:
prefer `select_related()`

Reply:
ok, let me try it out.

Reply:
I thought the select_related() is more appropriate when the model contains a foreign key?

Reply:
sorry, hadn't checked whether this model has foreign keys